File: Easy/buySweets/buySweets.c
#include <stdio.h>
#include <assert.h>

int main(void){
	int T,N,X,sum,f;

	scanf("%d",&T);
	assert(T>=1 && T<=20);

	while(T>0){
		scanf("%d %d",&N,&X);

		assert(N>=1 && N<=100);
		assert(X>=1 && X<=100);

		int A[N];
		sum = 0;
		f = 0;

		for(int i=0;i<N;i++){
			scanf("%d",&A[i]);
			assert(A[i]>=1 && A[i]<=100);
			sum += A[i];
		}

		for(int i=0;i<N;i++){
			if((sum-A[i])/X==sum/X){
				printf("-1\n");
				f = 1;
				break;
			}
		}

		if(f==0){
			printf("%d\n",sum/X);
		}

		T--;
	}
	
	return 0;
}

File: Easy/buySweets/buySweets.java
//run as java -ea buySweets
import java.util.Scanner;

public class buySweets{
	public static void main(String[] args){
		int T,N,X,sum,f;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=20) : "T must be 1 <= T <= 20";

		while(T>0){
			N = scan.nextInt();
			X = scan.nextInt();

			assert(N>=1 && N<=100) : "N must be 1 <= N <= 100";
			assert(X>=1 && X<=100) : "X must be 1 <= X <= 100";

			int[] A = new int[N];
			sum = 0;
			f = 0;

			for(int i=0;i<N;i++){
				A[i] = scan.nextInt();
				assert(A[i]>=1 && A[i]<=100) : "A["+i+"] must be 1 <= A[i] <= 100";
				sum += A[i];
			}

			for(int i=0;i<N;i++){
				if((sum-A[i])/X==sum/X){
					System.out.println(-1);
					f = 1;
					break;
				}
			}

			if(f==0){
				System.out.println(sum/X);
			}


			T--;
		}
	}
}